synphot/filter_parameterization/filter_fft.py

```diff
@@ -0,0 +1,233 @@
+
+    m = (np.sum([Sine1D(amplitude=fft_parameters[i].real / n_wave,
+                        frequency=i / n_wave, phase=0.25)
+                 for i in range(n_fft_pars)]) -
+         np.sum([Sine1D(amplitude=fft_parameters[i].imag / n_wave,
+                        frequency=i / n_wave)
+                 for i in range(n_fft_pars)]))
+
+    @custom_model
+    def fft_model(x):
+        """Approximate Fourier reconstruction of an astronomical filter.
+
+        Parameters
+        ----------
+        x : `~astropy.units.quantity.Quantity`
+            Full wavelength range that samples the filter.
+
+        Returns
+        -------
+        transmittance : array-like or `~astropy.units.quantity.Quantity`
+            Transmittance curve. If ``tr_max`` is a Quantity, this will
+            be a Quantity as well.
+
+        """
+        wave_unit = SpectralElement._internal_wave_unit
+        x = validate_quantity(x, wave_unit, equivalencies=u.spectral())
+
+        mo = m((x - wavelength.min()) / (wavelength[1] - wavelength[0]))
+        return (mo - mo.min()) * tr_max / mo.ptp()
+
+    return fft_model()
+
+
+def filters_to_fft_table(filters_mapping, n_terms=10):
+    """Run :func:`filter_to_fft` on a list of filters
+    and store results in a table.
+
+    Parameters
+    ----------
+    filters_mapping : dict
+        Dictionary mapping human-readable filter name to its
+        `~synphot.spectrum.SpectralElement` and wavelengths, if applicable.
+        If the filter object has a valid ``waveset``, just provide `None`
+        for wavelengths; otherwise provide a Quantity array for sampling.
+        For example::
+
+            {'JOHNSON/V': (<SpectralElement ...>, None),
+             'Flat': (<SpectralElement ...>, <Quantity [1000., ..., 9999.] Angstrom>)}
+
+    n_terms : int
+        Number of FFT parameters to keep.
+
+    Returns
+    -------
+    fft_table : `~astropy.table.Table`
+        Table storing FFT parameterization for the given filters.
+        Use its ``write`` method to save it to file.
+
+    """  # noqa
+    wave_unit = SpectralElement._internal_wave_unit
+    colnames = ['filter', 'n_lambda', 'lambda_0', 'delta_lambda',
+                'tr_max'] + [f'fft_{i}' for i in range(n_terms)]
+    rows = []
+
+    for key, (bp, wavelengths) in filters_mapping.items():
+        n_lambda, lambda_0, delta_lambda, tr_max, fft_pars = filter_to_fft(
+            bp, wavelengths=wavelengths, n_terms=n_terms)
+        if not NUMPY_LT_1_17:
+            rows.append(tuple(
+                [key, n_lambda, lambda_0, delta_lambda, tr_max] + fft_pars))
+        else:  # Numpy 1.16 cannot handle unit here
+            rows.append(tuple(
+                [key, n_lambda, lambda_0.value, delta_lambda.value,
+                 tr_max.value] + fft_pars))
+
+    fft_table = Table(rows=rows, names=colnames)
+    fft_table['lambda_0'].unit = wave_unit
+    fft_table['delta_lambda'].unit = wave_unit
+
+    return fft_table
```

Comment:
Maybe it also makes sense to add a convenience function `filter_table_to_models` function that takes a table and returns a dictionary of models for all of the table rows? This could also be a follow-on PR, though.

Reply:
Opened issue at #334